apps/electron/layers/preload/src/index.ts

```diff
@@ -30,6 +31,30 @@ contextBridge.exposeInMainWorld('apis', {
 
   onSidebarVisibilityChange: (visible: boolean) =>
     ipcRenderer.invoke('ui:sidebar-visibility-change', visible),
+  signIn: () => {
+    electronGoogleOauth.openAuthWindowAndGetTokens();
```

Comment:
We shall use the Firebase SDK, instead of plain Google Auth here.

Reply:
firebase sdk only can set `xx.firebaseapp.com/__/__auth/handler` as google auth callback handler.
so it can't redirect to localhost or affine.pro

Reply:
tried 

<img width="634" alt="image" src="https://user-images.githubusercontent.com/4948120/230411829-382f47b9-90d3-4456-bd18-ac3b087be552.png">

but

<img width="976" alt="image" src="https://user-images.githubusercontent.com/4948120/230411637-6157eb81-ff7c-4208-95c2-8905934fc4e9.png">